tools/chatgpt/chatgpt.py

```diff
@@ -39,34 +39,38 @@
 vision_sup_ext = ["jpg", "jpeg", "png", "webp", "gif"]
 
 file_search_file_streams = []
-image_urls = []
+image_files = []
 
 for path in context_files:
     ext = path.split(".")[-1].lower()
-    if ext in vision_sup_ext and model in ["gpt-4o", "gpt-4o-mini", "gpt-4-turbo"]:
+    if ext in vision_sup_ext:
+        if model not in ["gpt-4o", "gpt-4o-mini", "gpt-4-turbo"]:
+            raise Exception("Vision model is not supported for this model!")
         if os.path.getsize(path) > 20 * 1024 * 1024:
-            raise Exception(f"File {path} exceeds the 20MB limit and will not be processed.")
+            raise Exception(
+                f"File {path} exceeds the 20MB limit and will not be processed."
+            )
         file = client.files.create(file=open(path, "rb"), purpose="vision")
         promt = {"type": "image_file", "image_file": {"file_id": file.id}}
-        image_urls.append(promt)
-
+        image_files.append(promt)
     elif ext in file_search_sup_ext:
         file_search_file_streams.append(open(path, "rb"))
     else:
-        raise Exception("Not supported file!")
+        raise Exception(f"File type {ext} is not supported!")
```

Comment:
why can this even happen? We can define which filetypes are allowed in the UI

Reply:
It never happens, it is just for caution.